firmware/application/string_format.cpp

```diff
@@ -162,13 +162,47 @@ std::string to_string_decimal(float decimal, int8_t precision) {
     return result;
 }
 
+// right-justified frequency in Hz, always 10 characters
 std::string to_string_freq(const uint64_t f) {
-    auto final_str = to_string_dec_int(f / 1000000, 4) + to_string_dec_int(f % 1000000, 6, '0');
+    std::string final_str{""};
+
+    if (f < 1000000)
+        final_str = to_string_dec_int(f, 10, ' ');
+    else
+        final_str = to_string_dec_int(f / 1000000, 4) + to_string_dec_int(f % 1000000, 6, '0');
```

Comment:
I know this isn't your change, but why is this broken into two parts. I couldn't figure out why they did this instead of just printing the 64bit value directly (which is why I added the 64bit overload for to_string_dec_int).

Reply:
Is it just to make sure the output is a fixed width?

Reply:
I assume this is because the to_string_dec_int() function expects a 32-bit value.  I don't see your 64-bit overloaded version.

Reply:
https://github.com/eried/portapack-mayhem/blob/next/firmware/application/string_format.cpp 
Line 28, that algorithm works just fine for 64 bit. Could do the same for the function at 43 (use a template `Int` for the type of `n`) -- You'd have to also add the actual 64bit wrapper for   std::string to_string_dec_uint -- or that could probably even be templated.

Don't worry about it unless you want to, that whole file is due for some cleanup.

Reply:
I think I'll defer this one for today.  :-)